### ISSUE TITLE
Typescript definitions disallow valid ClientOptions

### ISSUE THREAD

**Issue by KieranP** [ably/ably-js#616]
I'm switching from raw JS to typescript. In doing so, the TS definitions for this project don't seem to allow the following:

```
let options : Ably.Types.ClientOptions = {
  key: '...',
  clientId: '...',
  closeOnUnload: true,
  disconnectedRetryTimeout: 1000,
  suspendedRetryTimeout: 1000
}

let ablyClient = new Ably.Realtime(options)
```

key and clientId are accepted, but closeOnUnload, disconnectedRetryTimeout, and suspendedRetryTimeout are not.

According to https://www.ably.io/documentation/realtime/usage,  those 3 are valid arguments to the Realtime constructor.

**Comment by paddybyers** [comment]
Thanks for reporting this - we'll have a look.

**Comment by SimonWoolf** [comment]
Fixed by https://github.com/ably/ably-js/commit/2e24fef28e791745fa8ee0ef40420a07d1a39390 ; released in ably-js 1.1.22
